packages/exoflex/src/components/Switch.tsx

```diff
@@ -22,6 +22,29 @@ export type SwitchProps = AccessibilityProps & {
 
 const MARGIN = 2;
 
+export function getTrueWidth(
+  type: 'track' | 'thumb',
+  defaultWidth: number,
+  style?: StyleProp<ViewStyle>,
+): number {
+  let flattenedStyle = StyleSheet.flatten(style);
+  let width = flattenedStyle?.width;
+  let minWidth = flattenedStyle?.minWidth;
+  let maxWidth = flattenedStyle?.maxWidth;
+  let trueWidth = !!width
+    ? +width
+    : type === 'track'
+    ? defaultWidth
+    : defaultWidth / 2;
+  if (!!minWidth) {
+    trueWidth = +minWidth > trueWidth ? +minWidth : trueWidth;
+  }
+  if (!!maxWidth) {
+    trueWidth = +maxWidth < trueWidth ? +maxWidth : trueWidth;
+  }
+  return trueWidth;
```

Comment:
Hmm, I think it's better if we extract this part out to something like `function clamp(number, upper, lower): number` so it's easier to read and maintain.

Then we can use it like:
```ts
return clamp(trueWidth, maxWidth, minWidth);
```

Reply:
Something like this:
https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L13981

Reply:
Interesting! I also found the simpler version here
https://github.com/hughsk/clamp/blob/377851f0cca9f3f134b53881e294782cccdae4d8/index.js

Reply:
Interesting!